csprng.php

```diff
@@ -62,15 +86,18 @@ function random_bytes($bytesLength)
 function random_hex($stringLength)
 {
     // @todo Validate $stringLength here
+    if ($stringLength < 1) {
+        throw new InvalidArgumentException('random_bytes() expects byte length greater than 0.');
+    }
 
     // @todo Double check bin2hex() string length calculations
-    $length = (int) round($stringLength / 2);
-    $bytes = random_bytes($length);
+    $randomLength = ceil($stringLength / 2);
+    $bytes = random_bytes($randomLength);
 
     $asHex = bin2hex($bytes);
 
-    // @todo Not sure of mb_string support across systems
-    return mb_substr($asHex, 0, $stringLength);
+    // mb_substr is not needed on ASCII strings
```

Comment:
The `random_hex()` function didn't get any love and was [removed from the patch](https://github.com/SammyK/php-src/commit/a1e62293b12f4440da55846fd50cae37b4c53265).